libs/coin-modules/coin-casper/src/api/types.ts

```diff
@@ -109,5 +114,10 @@ export interface NAccountInfo {
 
 export interface NDeployMessagePutResponse {
   api_version: string;
-  deploy_hash: string;
+  error?: {
+    code: number;
+    message: string;
+    data: string;
+  };
+  deploy_hash?: string;
```

Comment:
This could be 

```
export type NDeployMessagePutResponse {
    api_version: string;
    deploy_hash: string;
 }
 
 export type NDeployMessagePutError {
    api_version: string;
    error?: {
       code: number;
       message: string;
       data: string;
     }
  };
  
  export type NDeployMessagePut = NDeployMessagePutResponse | NDeployMessagePutError
  
  ```

Reply:
Great catch! The type was actually wrong, added it during testing but the error always only appear on the root response.